pkg/controller/vitessshardreplication/reconcile_drain.go

```diff
@@ -313,11 +313,10 @@ func (r *ReconcileVitessShard) reconcileDrain(ctx context.Context, vts *planetsc
 	return resultBuilder.Result()
 }
 
-func (r *ReconcileVitessShard) handleExternalReparent(ctx context.Context, vts *planetscalev2.VitessShard, wr *wrangler.Wrangler, newMasterTablet *topodatapb.Tablet, oldMasterAlias *topodatapb.TabletAlias) error {
-	err := wr.TabletManagerClient().TabletExternallyReparented(ctx, newMasterTablet, "")
+func (r *ReconcileVitessShard) handleExternalReparent(ctx context.Context, vts *planetscalev2.VitessShard, wr *wrangler.Wrangler, newMasterAlias, oldMasterAlias *topodatapb.TabletAlias) error {
+	err := wr.TabletExternallyReparented(ctx, newMasterAlias)
 
 	if err == nil {
-
 		// TODO: Create a specialized function to take the place of TER, that sets the old master as spare, rather than
```

Comment:
You could use the `demote_master_type=SPARE` vttablet flag to fix this TODO.

Reply:
Ooh nice! Thanks!